lib/Elevate/Database.pm

```diff
@@ -75,4 +87,64 @@ sub get_db_info_if_provided_by_cloudlinux ( $use_cache = 1 ) {
     return ( $db_type, $db_version );
 }
 
+sub get_local_database_version () {
+
+    my $version;
+
+    eval {
+        local $Cpanel::MysqlUtils::Version::USE_LOCAL_MYSQL = 1;
+        $version = Cpanel::MysqlUtils::Version::uncached_mysqlversion();
+    };
+    if ( my $exception = $@ ) {
+        WARN("Error encountered querying the version from the database server: $exception");
+
+        # Load it from the configuration if we cannot get the version directly
+        my $cpconf = Cpanel::Config::LoadCpConf::loadcpconf();
+        $version = $cpconf->{'mysql-version'};
+    }
+
+    return $version;
+}
+
+sub is_database_version_supported ($version) {
+
+    return scalar grep { $version eq $_ } SUPPORTED_MYSQL_VERSIONS;
+}
+
+sub get_default_upgrade_version () {
+
+    require Whostmgr::Mysql::Upgrade;
+
+    return Whostmgr::Mysql::Upgrade::get_latest_available_version( version => get_local_database_version() );
+}
+
+sub get_database_type_name_from_version ($version) {
+    return Cpanel::MariaDB::version_is_mariadb($version) ? 'MariaDB' : 'MySQL';
+}
+
+sub validate_mysql_upgrade_version ($upgrade_version) {
```

Comment:
FIX:

This would still allow something like upgrading MySQL 5.6 to MySQL 5.7 which would cause an elevate failure.  Additionally, after the initial check from lines 130-145 completes, we should run the version through `Elevate::Database::is_database_version_supported` to validate that the mysql target is an eligible upgrade target for elevate too.  In other words, 

1.  Confirm that cPanel can/will do the upgrade
2. Confirm that the version is valid for elevate